pkg/apis/jenkins.io/v1/types_fact.go

```diff
@@ -0,0 +1,99 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// Fact represents observed facts
+type Fact struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec   FactSpec   `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	Status FactStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// FactSpec is the specification of a Fact
+type FactSpec struct {
+	Name             string        `json:"name" protobuf:"bytes,1,opt,name=name"`
+	ID               int           `json:"id" protobuf:"bytes,2,opt,name=id"`
+	FactType         string        `json:"factType" protobuf:"bytes,3,opt,name=factType"`
+	Measurements     []Measurement `json:"measurements" protobuf:"bytes,4,opt,name=measurements"`
+	Statements       []Statement   `json:"statements" protobuf:"bytes,5,opt,name=statements"`
+	Original         Original      `json:"original,omitempty" protobuf:"bytes,6,opt,name=original"`
+	Tags             []string      `json:"tags,omitempty" protobuf:"bytes,7,opt,name=tags"`
+	SubjectReference string        `json:"subject" protobuf:"bytes,8,opt,name=source"`
+}
+
+// FactStatus is the status for an Environment resource
+type FactStatus struct {
+	Version string `json:"version,omitempty" protobuf:"bytes,1,opt,name=version"`
+}
+
+// Measurement contains the value measured on this fact
+type Measurement struct {
+	Name             string   `json:"name" protobuf:"bytes,1,opt,name=name"`
+	MeasurementType  string   `json:"measurementType" protobuf:"bytes,2,opt,name=measurementType"`
+	MeasurementValue int      `json:"measurementValue" protobuf:"bytes,3,opt,name=measurementValue"`
+	Tags             []string `json:"tags,omitempty" protobuf:"bytes,4,opt,name=tags"`
+}
+
+// Statement is the Fact statement
```

Comment:
Not very useful documentation. What is a “statement”? How does it differ from a “measurement”? When would you use it? When extracting the new CRD we should take the opportunity to review all the existing fields and think carefully about what they are needed for and why, and record that documentation.

Reply:
Agree. @pmuir @snoopdave can you provide a clear definition of the different fields present in the Fact CRD? I've read the Core3 data model, but I'm still not sure how to define them.

Reply:
I added the definitions found on the Core3 Data model, but I think they can be improved.

Reply:
One for @snoopdave

Reply:
from the original design a Measurement is different from a Statement. A a measurement is a percentage or a count; something measured. A Statement represents a decision that was made, for example that a promition was approved or denied.

Reply:
I'll update the doc in the CRD with the following comments:

```
// Measurement is a percentage or a count, something measured that the system will capture within a fact
```

and

```
// Statement represents a decision that was made, for example that a promotion was approved or denied
```

Btw @snoopdave , what about `Original`? Should we change it's structure? @jglick was saying that current implementation is not really compatible with having several reports.